packages/test/test-end-to-end-tests/src/test/mapEndToEndTests.spec.ts

```diff
@@ -374,6 +381,74 @@ describeNoCompat("SharedMap orderSequentially", (getTestObjectProvider) => {
             clearEventCount++;
         });
     });
+    describe("Concurrent op processing", () => {
+        let container2: Container;
+        let dataObject2: ITestFluidObject;
+        let sharedMap2: SharedMap;
+
+        beforeEach(async () => {
+            await reset();
+            provider = getTestObjectProvider();
+        });
+
+        afterEach(async () => reset());
+
+        const setupContainers = async (
+            containerConfig: ITestContainerConfig,
+            featureGates: Record<string, ConfigTypes> = {},
+        ) => {
+            const configWithFeatureGates = {
+                ...containerConfig,
+                loaderProps: { configProvider: configProvider(featureGates) },
+            };
+            const container1 = await provider.makeTestContainer(configWithFeatureGates) as Container;
+            container2 = await provider.loadTestContainer(configWithFeatureGates) as Container;
+
+            dataObject = await requestFluidObject<ITestFluidObject>(container1, "default");
+            dataObject2 = await requestFluidObject<ITestFluidObject>(container2, "default");
+
+            sharedMap = await dataObject.getSharedObject<SharedMap>(mapId);
+            sharedMap2 = await dataObject2.getSharedObject<SharedMap>(mapId);
+
+            await provider.ensureSynchronized();
+        };
+
+        itExpects("Should close container when sending an op while processing another op",
+            [{
+                eventName: "fluid:telemetry:Container:ContainerClose",
+                error: "Making changes to data model is disallowed while processing ops.",
+            }], async () => {
+                await setupContainers(testContainerConfig, { "Fluid.Container.ConcurrentOpSend": true });
+
+                sharedMap.on("valueChanged", (changed, local) => {
+                    if (!local) {
+                        assert.equal(changed.key, "key2", "Incorrect value for key1 in container 1");
+                    }
+                    // Avoid re-entrancy by setting a new key
+                    if (changed.key !== "key2") {
+                        sharedMap2.set("key2", "v2");
```

Comment:
This test may be over-complicated?  Not sure, I'm not confident I'm reading it right.  What it looks like to me:

- There are two container instances loaded by the same loader, so using the same DeltaManager.
- This test is simulating the 2nd container firing an op while processing an op from the first container.

Why do you need 2 containers?  And this seems like an unlikely scenario to simulate.  Closest thing I can imagine would be like app code in the main container somehow triggering something that causes that Summarizer to submit an op synchronously.  Maybe it's worth covering that, but maybe not.

I guess now that I write all that out... no harm in testing across that dimension, I was just surprised. So feel free to leave as-is.